Add a Gym/Gymnasium environment for all quadruped systems.

Danfoa opened the pull request:
This PR introduces a Gymnasium environmet to expose the simulation and control problem of quadrupeds with a standard API. 

This environment should be flexible enough to enable user to define

1. Which robot, terrain, disturbance distribution, and domain randomization to use.
2. Which type of base velocity commands (linear and angular velocity) should the robot try to track. 
3. Compare different control policies in terms of energy efficiency and robustness 
4. Generate dataset of observations of state and actions for imitation learning purposes.

giulioturrisi commented:
Hi Daniel, nice work!

I found some errors running the code. For example, opimize_swing variable should be set somewhere globally (otherwise the sampling with gait adaptation does not work). Plus, i have some error running the RTI methods, and the plot of the swing are missing.

giulioturrisi commented:
Hi Daniel, nice work!

I found some errors running the code. For example, opimize_swing variable should be set somewhere globally (otherwise the sampling with gait adaptation does not work). Plus, i have some error running the RTI methods, and the plot of the swing are missing.

Danfoa commented:
Hi @giulioturrisi 

Added the plotting of the feet trajectories in the simulation script, I believe we should keep all "non-generic" functions outside of the QuadrupedEnv class, to keep it clean. 

I modify the plotting funciton and introduced some utility rendering functions for visualizing some geometric primitives for debugging. 

I think the feet trajectory function can be further improved but I leave this to you guys.

Danfoa commented on simulation/simulation_mujoco.py:
```diff
@@ -305,578 +197,338 @@
+    # Update the periodic gait generator
+    pgg.run(simulation_dt, pgg.step_freq)
+
+    if env.step_num > 2000:
```
@giulioturrisi 

Here Giulio if I reset the environment the controller sometimes goes to hell. I think there are control specific variables that need to be resetted/reinitialized. Can you have a look please as you are more familiar with the controller?

Also it would be ideal if in the Controller class which will cover all posible controllers there is a .reset() method to deal with this in an Object Oriented fashion.
